### ISSUE TITLE
addType resets overlays

### ISSUE THREAD

**Issue by Deklin** [sporritt/jsPlumb#311]
I tested the patch on both 1.7.3. alpha as well as pulled the change for 288 and unfortunately the issue still exists.

What is happening is when you use addType on a connection, the overlay is reset back to default. 

If its a custom overlay, its reset, if its a label overlay the label value is blown away.

A simple way to test this is 
Following the unit test here;
https://github.com/sporritt/jsPlumb/commit/bc1a450b430dba6b7bab9bfb556239cccb209100

        ok(e.getOverlay("myLabel1") != null, "label overlay was retrieved");
+
+        e.addType("selected");
+        ok(e.getOverlay("myLabel1") != null, "label overlay was not blown away");
+        e.removeType("selected");
+        ok(e.getOverlay("myLabel1") != null, "label overlay was not blown away");

This is not in fact destroying the label, instead do this

var labelOverlay;
labelOverlay = e.getOverlay("myLabel1");
labelOverlay.setLabel("foo");
ok(labelOverlay.getLabel() === "foo");
e.addType("selected");
labelOverlay = e.getOverlay("myLabel1");
ok(labelOverlay.getLabel() === "foo"); // This will fail as label was reset

If you perform the same test with a custom overlay, the entire overlay is also reset back to its initial 'create' state.

**Comment by sporritt** [comment]
tracking this in the [original issue](https://github.com/sporritt/jsPlumb/issues/288)

**Comment by sporritt** [comment]
```
var labelOverlay;
labelOverlay = e.getOverlay("myLabel1");
labelOverlay.setLabel("foo");
ok(labelOverlay.getLabel() === "foo");
e.addType("selected");
labelOverlay = e.getOverlay("myLabel1");
ok(labelOverlay.getLabel() === "foo"); // This will fail as label was reset
```

**Comment by Deklin** [comment]
Ok thanks for the clarification

**Comment by sporritt** [comment]
no problem.  this might actually be a bit fiddly to fix, unfortunately.

**Comment by Deklin** [comment]
I have worked around it by not using addType, however my workaround is.. ugly.
:)

var overlay = $(item.canvas)

and since overlay is SVG
I can't use $.addClass or removeClass, so I use $.attr to add/remove the attr I need. One advantage is I can use CSS now to deal w/ the styles, however the problem is using CSS I am guessing when to stroke, vs fill based on assuming child 1 = ignore, child 2= string, child 3 = arrow

ugly unfortunately

**Comment by sporritt** [comment]
targeting 1.7.3 for this.  the types stuff was not originally conceived of as something you'd call very often, but in an application i am working on i find myself using it for something a bit like hover states, which is cool, but which leaks memory like a sieve: say you have a type that has an anchor and a connector defined. every time you activate that type, a new instance of the anchor and connector are created.  not awesome.

**Comment by sporritt** [comment]
this is fixed in 1.7.3 dev.
